nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/ListAzureDataLakeStorage.java

```diff
@@ -226,9 +239,12 @@ protected String getPath(ProcessContext context) {
             options.setRecursive(recurseSubdirectories);
 
             Pattern baseDirectoryPattern = Pattern.compile("^" + baseDirectory + "/?");
+            final String tempFilePrefix = defaultIfBlank(System.getProperty("tempFilePrefix"), TEMP_FILE_PREFIX);
```

Comment:
Direct calls to `System.getProperty()` should be avoided. Can you explain the reason for getting the value of a system property?

Reply:
@exceptionfactory thanks for your remarks. The idea was to help providing a fix quickly in case the Customer needs to change the prefix for example they are already using the prefix or the path would be too long because of the additional characters of the prefix. Is there any other way the prefix could be changed without a property or code changing? I am open for ideas.

Reply:
Thanks for the clarification @timeabarna.

It seems like there are multiple competing use cases here, so thank you for working through the feedback.

On the one hand, the purpose of the Temporary File Prefix seems to be something internal to NiFi so that the PutAzureDataLakeStorage can write files with a temporary name that ListAzureDataLakeStorage will not pick up. For that use case, having an internal shared value, that cannot be changed, seems like the most logical approach.

One the other hand, if other components or services outside of NiFi may write files that could be considered temporary, then perhaps more configuration is necessary using a standard processor property descriptor. However, since ListAzureDataLakeStorage already has a `File Filter` property, it seems like that could be used to exclude temporary files from other sources.

If the goal is to support maximum configurability in `PutAzureDataLakeStorage` when writing temporary files, then it should be a processor property. However, if the temporary file prefix is specific enough to NiFi, using something like `AzureDataLakeStorage.` or perhaps prefixed with a period as `.AzureDataLakeStorage.`, then that seems like it should cover the intended use cases without needing to be configured.

Perhaps @jfrazee has some additional thoughts.

Reply:
I don't think there's much precedent for exposing config to processors through system properties. I can definitely see how it'd be convenient, but we can achieve this with the file filter so it seems redundant here.

Reply:
@exceptionfactory @timeabarna There's temp file handling to be considered in both the List and the Put so wanted to summarize here as well. IMO, List should probably filter _ and . by default, but it's configurable so still up to the user. The temp files written by Put should probably default to being written in a hidden directory and have random prefixes, but with the directory configurable so it could potentially be not in the same directory.

What do you think?

Reply:
@jfrazee @exceptionfactory Let me summarise my understanding on your recommendations as it seems that the final solution would be a mix of the old things :)
List processor:
- Modify listing to hide any file and folder which names starting with either _ or .
- Keep the boolean property which is turning on the filtering of the above

Put processor:
- Create a property for tempfolder . 
- Modify rename to put the temp file in the provided tempfolder under a generated UUID with the existing file name

My questions:
1. As renaming will be location related not name related, is this still atomic?
2. Should tempfolder property used as name only, location only or both? This requires different validation.
3. What happens if a Customer would like to use Put processor as in PutHDFS, which is putting the temp file in the same directory as the file is located, using . + filename as naming? Should we allow this?

Reply:
The renaming should still be atomic.

Are you asking if we should support a temporary name prefix and a temporary directory? I'd go with one or another and not both. I prefer a location because in a high volume ingest scenario there could be a lot of temporary files.

I'd have to look closer at PutHDFS. I don't think that's ideal behavior though if there's a possibility of multiple writers.